src/main/java/Duke.java

```diff
@@ -1,10 +1,216 @@
+    private Parser parser;
+
+    /**
+     * Duke class constructor for JavaFX.
+     */
+    public Duke() {
+        ui = new Ui();
+        storage = new Storage();
+        tasks = new TaskList(storage.readFile());
+        Scanner sc = new Scanner(System.in);
+        parser = new Parser(sc);
+    }
+
+    /**
+     * Main method that drives the Duke bot, using Command Line.
+     */
+    public void run() {
+        Scanner sc = new Scanner(System.in);
+        Parser parser = new Parser(sc);
+        ui.printHello();
+
+        String command = parser.readCommand();
+        try {
+            while (!parser.getCommandType(command).equals("bye")) {
+
+                String commandType = parser.getCommandType(command);
+
+                switch (commandType) {
+                case "list":
+                    tasks.printList();
+                    command = parser.readCommand();
+                    break;
+                case "find":
+                    String findString = parser.getFind(command);
+                    TaskList matchedTask = tasks.find(findString);
+                    ui.printFind(matchedTask);
+                    command = parser.readCommand();
+                    break;
+                case "done": {
+                    int taskNo = parser.getTaskNo(command);
+                    try {
+                        tasks.setDone(taskNo);
+                        ui.printDone(tasks.getTask(taskNo));
+                    } catch (IndexOutOfBoundsException e) {
+                        System.out.println("There is no task " + (taskNo + 1) + " in the list.");
+                    }
+                    command = parser.readCommand();
+                    break;
+                }
+                case "delete": {
+                    int taskNo = parser.getTaskNo(command);
+                    Task task = tasks.getTask(taskNo);
+                    ui.printDelete(task, tasks.getSize() - 1);
+                    tasks.deleteTask(taskNo);
+                    storage.saveFile(tasks.getTaskList());
+                    command = parser.readCommand();
+                    break;
+                }
+                case "deadline":
+                    try {
+                        Deadline deadline = parser.getDeadline(command);
+                        tasks.addTask(deadline);
+                        storage.saveFile(tasks.getTaskList());
+                        ui.printAdd(deadline, tasks.getSize());
+                    } catch (DukeException | ParseException e) {
+                        System.out.println(e.getMessage());
+                    }
+                    command = parser.readCommand();
+                    break;
+                case "event":
+                    try {
+                        Event event = parser.getEvent(command);
+                        tasks.addTask(event);
+                        storage.saveFile(tasks.getTaskList());
+                        ui.printAdd(event, tasks.getSize());
```

Comment:
You can abstract out the repeated part of code into a method, or put this part to another class like `parser` to make the code cleaner.